sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -53,6 +56,10 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
       tableIdent.table.toLowerCase)
   }
 
+  /** ReadWriteLock for each tables, protect the read and write cached */
+  protected[hive] val tableLockMap =
+    new ConcurrentHashMap[QualifiedTableName, ReentrantReadWriteLock]
```

Comment:
https://google.github.io/guava/releases/19.0/api/docs/com/google/common/util/concurrent/Striped.html is usually good enough